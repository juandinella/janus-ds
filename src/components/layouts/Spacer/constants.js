export const options = {
  /**
   * Sizes of the spacer
   */
  sizes: ['0', '4', '8', '12', '16', '24', '32', '48', '56', '64'],
}
