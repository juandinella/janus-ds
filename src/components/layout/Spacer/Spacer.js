import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import Horizontal from './components/Horizontal/Horizontal'
import Vertical from './components/Vertical/Vertical'
import { options } from './constants'
import styles from './Spacer.styles'

/**
 * Provides empty space.
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledSpacer = styled.div.withConfig({ shouldForwardProp })`
  ${styles}
`

const Spacer = ({ id, mobileSize, size = '4', isPlayground = false }) => (
  <StyledSpacer
    id={id}
    data-testid={id}
    size={size}
    mobileSize={mobileSize}
    isPlayground={isPlayground}
  />
)

Spacer.displayName = 'Spacer'

Spacer.propTypes = {
  /** Differents sizes of the spacer */
  size: PropTypes.oneOf(options.sizes),
  /* Size for mobile */
  mobileSize: PropTypes.oneOf(options.sizes),
  /** This Boolean attribute add border to see it on Storybook */
  isPlayground: PropTypes.bool,
  /** Spacer's id */
  id: PropTypes.string,
}

Spacer.Horizontal = Horizontal
Spacer.Vertical = Vertical

export default Spacer
