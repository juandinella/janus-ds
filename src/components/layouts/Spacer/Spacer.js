import React from 'react'
import PropTypes from 'prop-types'
import styles from './Spacer.styles'
import Horizontal from './components/Horizontal/Horizontal'
import Vertical from './components/Vertical/Vertical'
import { options } from './constants'

/**
 * Provides empty space.
 */

const Spacer = ({ id, mobileSize, size, isPlayground }) => (
  <div
    css={styles}
    id={id}
    data-testid={id}
    size={size}
    mobileSize={mobileSize}
    isPlayground={isPlayground}
  />
)

Spacer.displayName = 'Spacer'

Spacer.propTypes = {
  /** Differents sizes of the spacer */
  size: PropTypes.oneOf(options.sizes),
  /* Size for mobile */
  mobileSize: PropTypes.oneOf(options.sizes),
  /** This Boolean attribute add border to see it on Storybook */
  isPlayground: PropTypes.bool,
  /** Spacer's id */
  id: PropTypes.string,
}

Spacer.defaultProps = {
  size: '4',
  isPlayground: false,
}

Spacer.Horizontal = Horizontal
Spacer.Vertical = Vertical

export default Spacer
