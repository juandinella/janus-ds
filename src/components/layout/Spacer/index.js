import Spacer from './Spacer'

export default Spacer
export { options } from './constants'
