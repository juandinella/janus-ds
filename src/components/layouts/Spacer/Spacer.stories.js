import { options } from './constants'
import Spacer from './Spacer'

import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../../helpers/storybook'

const Template = getTemplate(Spacer)
const ListTemplate = getListTemplate(Spacer)

export default {
  title: 'Layout/Spacer',
  component: Spacer,
  args: {
    isPlayground: true,
  },
  argTypes: {
    ...getOptionsArgTypes(options.sizes, 'size'),
    ...getOptionsArgTypes(options.sizes, 'mobileSize'),
  },
  parameters: { __sb: { fd: 'row' } },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }
