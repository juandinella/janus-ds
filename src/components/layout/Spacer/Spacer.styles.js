import { css } from 'styled-components'
import layout from '../../../styles/theme/layout'

export default css`
  ${(props) =>
    props.size &&
    css`
      min-width: ${(props) =>
        props.mobileSize ? `${props.mobileSize}px` : `${props.size}px`};
      min-height: ${(props) =>
        props.mobileSize ? `${props.mobileSize}px` : `${props.size}px`};

      @media screen and (min-width: ${() => layout.breakpoints.sm}) {
        width: ${(props) => `${props.size}px`};
        height: ${(props) => `${props.size}px`};
        min-width: ${(props) => `${props.size}px`};
        min-height: ${(props) => `${props.size}px`};
      }
    `}

  ${(props) =>
    props.isPlayground &&
    css`
      border: 2px dashed;
      border-color: var(--color-neutral-900);
    `}
`
