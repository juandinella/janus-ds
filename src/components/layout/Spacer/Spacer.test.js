import initStoryshots from '@storybook/addon-storyshots'
import { cleanup, render, screen } from '@testing-library/react'
import React from 'react'
import Spacer from './Spacer'

initStoryshots({
  storyKindRegex: /^Layout\/Spacer$/,
})

const componentProps = {
  id: 'spacer',
  size: '24',
}

afterEach(cleanup)

describe('<Spacer />', () => {
  test('should render component correctly', () => {
    render(<Spacer {...componentProps} />)
    const spacer = screen.getByTestId(componentProps.id)
    expect(spacer).toBeInTheDocument()
  })

  test('should render horizontal correctly', () => {
    render(<Spacer.Horizontal {...componentProps} />)
    const spacer = screen.getByTestId(componentProps.id)
    expect(spacer).toBeInTheDocument()
  })

  test('should render vertical correctly', () => {
    render(<Spacer.Vertical {...componentProps} />)
    const spacer = screen.getByTestId(componentProps.id)
    expect(spacer).toBeInTheDocument()
  })

  test('should render component with correct size', () => {
    render(<Spacer {...componentProps} />)
    const spacer = screen.getByTestId(componentProps.id)
    expect(spacer).toHaveAttribute('size', `${componentProps.size}`)
  })
})
