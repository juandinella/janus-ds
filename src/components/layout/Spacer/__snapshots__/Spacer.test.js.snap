// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Spacer Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-ofq5q9-0 kjKZqh"
    size="4"
  />
</div>
`;

exports[`Storyshots Layout/Spacer Sizes 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-ofq5q9-0 dfPMsV"
    size="0"
  />
  <div
    className="sc-ofq5q9-0 kjKZqh"
    size="4"
  />
  <div
    className="sc-ofq5q9-0 eiAiKN"
    size="8"
  />
  <div
    className="sc-ofq5q9-0 eAFszd"
    size="12"
  />
  <div
    className="sc-ofq5q9-0 llvzih"
    size="16"
  />
  <div
    className="sc-ofq5q9-0 hFdNLd"
    size="24"
  />
  <div
    className="sc-ofq5q9-0 dPcXRR"
    size="32"
  />
  <div
    className="sc-ofq5q9-0 rHOoF"
    size="48"
  />
  <div
    className="sc-ofq5q9-0 fXAmsp"
    size="56"
  />
  <div
    className="sc-ofq5q9-0 criAVl"
    size="64"
  />
</div>
`;
